parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/BrokerPollSystem.java

```diff
@@ -136,6 +141,18 @@ private boolean controlLoop() {
                         doClose();
                     }
                 }
+
+                if (state == State.paused) {
```

Comment:
Also note that, you must continue to poll the broker otherwise you'll be kicked out of the group as the broker cannot determine if you're dead, or too slow.

Reply:
You're right, that's indeed an issue. So the best solution is probably indeed to not let the `BrokerPollSystem` know at all that the processor is paused and let it continue and finally pause the subscription once the buffers are full.